packages/client-search/src/types/Hit.ts

```diff
@@ -0,0 +1,47 @@
+type HighlightMatch = {
+  readonly value: string;
+  readonly matchLevel: 'none' | 'partial' | 'full';
+  readonly matchedWords: readonly string[];
+  readonly fullyHighlighted?: boolean;
+};
+
+type Highlight<THit> = THit extends string | number
+  ? HighlightMatch
+  : {
+      [KAttribute in keyof THit]: Highlight<THit[KAttribute]>;
+    };
+
+type SnippetMatch = {
+  readonly value: string;
+  readonly matchLevel: 'none' | 'partial' | 'full';
+};
+
+type Snippet<THit> = THit extends string | number
+  ? SnippetMatch
+  : {
+      [KAttribute in keyof THit]: Snippet<THit[KAttribute]>;
+    };
+
+export type Hit<THit> = THit & {
+  readonly objectID: string;
+  readonly _highlightResult?: Highlight<THit>;
+  readonly _snippetResult?: Snippet<THit>;
```

Comment:
I'm not sure, but maybe change the name of the type to HighlightResult and export it?

Reply:
See f3efb64080ded7a67aecf8b576523e84595c4906 and 82d6f8e98ddb2987daf6ca8b8db829b988dd5f12.